docs/sources/srv-record.md

```diff
@@ -0,0 +1,29 @@
+# SRV record with CRD source
+
+You can create and manage SRV records with the help of [CRD source](../contributing/crd-source.md)
+and `DNSEndpoint` CRD. The implementation of this feature depends on provider API support, this feature is currently known to be supported (at least) by `akamai`, `civo`, `cloudflare`, `ibmcloud`, `linode`, `rfc2136` and `pdns` providers.
+
+In order to start managing MX records you need to set the `--managed-record-types SRV` flag.
+
+```console
+external-dns --source crd --provider {akamai|civo|cloudflare|ibmcloud|linode|rfc2136|pdns} --managed-record-types A --managed-record-types CNAME --managed-record-types SRV
+```
+
+Targets within the CRD need to be specified according to the RFC 2782. Below is an example of
+`example.com` DNS SRV record. It specifies a `sip` service of `udp` protocol. It has two targets
+of identical priority but with different weights. They point to different backend servers.
+
+```yaml
+apiVersion: externaldns.k8s.io/v1alpha1
+kind: DNSEndpoint
+metadata:
+  name: examplesrvrecord
+spec:
+  endpoints:
+    - dnsName: _sip._udp.example.com
+      recordTTL: 180
+      recordType: SRV
+      targets:
```

Comment:
Would it be possible to share smoke tests as well, could you please also provide a way to test this manually with manifests and kubectl commands?  Similar to this example https://github.com/kubernetes-sigs/external-dns/pull/5111#issuecomment-2682090428

for cloudflare is enough, I'll do for aws or google?

I'm unsure where or not multiple targets supported, never tried that case.